File: data/code_pairs/fortran/1677.f90
! Fortran program to demonstrate file operations
program fileOpsFortran
  implicit none
  integer :: iounit
  character(len=100) :: line

  ! Create and open a file for writing
  open(unit=iounit, file='fortran_testfile', status='replace', action='write')
  ! Write a line to the file
  write(iounit,*) 'Test'
  ! Close the file
  close(iounit)

  ! Open the file for reading
  open(unit=iounit, file='fortran_testfile', status='old', action='read')
  ! Read the line from the file
  read(iounit,*) line
  ! Print the line read from the file
  print *, "Read from file:", line
  ! Close the file
  close(iounit)

  ! Delete the file
  call execute_command_line('rm fortran_testfile')
end program fileOpsFortran

File: data/code_pairs/cpp/1677.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cstdio> // For std::remove

int main() {
    std::string filename = "cpp_testfile";
    std::ofstream outFile(filename);

    // Check if file is open
    if (!outFile.is_open()) {
        std::cerr << "Failed to open the file for writing." << std::endl;
        return 1;
    }

    // Write to the file
    outFile << "Test" << std::endl;
    // Close the file
    outFile.close();

    // Open the file for reading
    std::ifstream inFile(filename);
    if (!inFile.is_open()) {
        std::cerr << "Failed to open the file for reading." << std::endl;
        return 1;
    }

    // Read from the file
    std::string line;
    std::getline(inFile, line);
    // Print the line read from the file
    std::cout << "Read from file: " << line << std::endl;
    // Close the file
    inFile.close();

    // Delete the file
    if (std::remove(filename.c_str()) != 0) {
        std::cerr << "Error deleting the file." << std::endl;
    }

    return 0;
}